### PR TITLE
Feature/getApproved task

### PR DESCRIPTION
**Pull request by AnnRiera**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,8 +1,4848 @@
 {
   "name": "harmony-marketplace-sdk",
   "version": "1.0.0",
-  "lockfileVersion": 1,
```
Comment (bzuker) [inline package-lock.json]: same as the other PR, I don't think there should be a package lock diff

File: src/hrc721.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { Contract } from '@harmony-js/contract'
 import { Harmony } from '@harmony-js/core'
 import { BigNumber, logger } from 'ethers'
 import { Logger } from 'ethers/lib/utils'
-import { BaseHRC721 } from './base-hrc721'
+import { BaseHRC721 } from './base-HRC721'
```
Comment (bzuker) [inline src/hrc721.ts:5]: why the name change? I think we are using `base-hrc721`

File: src/hrc721.ts [hunk 3]
```diff
@@ -30,7 +30,7 @@ export class HRC721 extends BaseHRC721 {
 
   async balanceOf(address: string): Promise<number> {
     if (!address) {
-      throw new Error('You have to provide an address')
+      throw new Error('You must to provide an address')
```
Comment (bzuker) [inline src/hrc721.ts:33]: nit: remove the `to`: `You must provide an address`

File: src/hrc721.ts [hunk 4]
```diff
@@ -62,7 +62,19 @@ export class HRC721 extends BaseHRC721 {
   }
 
   async getApproved(tokenId: string): Promise<string> {
-    return ''
+    if (!tokenId) {
+      throw new Error('You must to provide a tokenId')
```
Comment (bzuker) [inline src/hrc721.ts:66]: same as above `You must provide a tokenId`

File: package-lock.json (hunk 1 above)
Comment (AnnRiera) [reply package-lock.json]: Copy that!

File: src/hrc721.ts (hunk 2 above)
Comment (AnnRiera) [reply src/hrc721.ts:5]: It's because I changed the name (with uppercase only) 😅

File: src/hrc721.ts (hunk 4 above)
Comment (AnnRiera) [reply src/hrc721.ts:66]: Got it!

File: src/hrc721.spec.ts [hunk 5]
```diff
@@ -47,15 +47,15 @@ describe('HarmonyProvider', () => {
   })
 
   describe('ownerOf', () => {
-    it('should returns the owner of the tokenId token.', async () => {
+    it('should returns the owner of the tokenId token', async () => {
```
Comment (bzuker) [inline src/hrc721.spec.ts:50]: nit: `should return` instead of "returns", same in other tests below

File: src/hrc721.spec.ts [hunk 6]
```diff
@@ -71,4 +71,18 @@ describe('HarmonyProvider', () => {
...
+      expect(approved).to.exist
+      expect(approved).to.not.be.null
+      expect(approved).to.not.be.undefined
+      expect(approved).to.be.equals('0x0000000000000000000000000000000000000000')
+    })
+
+    it('should throw an error if tokenId is not provided', async () => {
+      expect(provider.getApproved('')).to.be.rejectedWith(Error)
+    })
+  })
```
Comment (bzuker) [inline src/hrc721.spec.ts:87]: let's leave a TODO here so that we can add more tests when we have the approve function working

**Review by bzuker — COMMENTED** [review]
two minor nits, then LGTM

File: src/hrc721.spec.ts (hunk 6 above)
Comment (AnnRiera) [reply src/hrc721.spec.ts]: Done!

File: src/hrc721.spec.ts (hunk 5 above)
Comment (AnnRiera) [reply src/hrc721.spec.ts]: Right, fixed!

**Review by bzuker — APPROVED** [review]
LGTM!

**Review by jarcodallo — APPROVED** [review]

**Pull request merged**